framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3442,4 +3528,26 @@ public ProcessingEnvironment getProcessingEnv() {
     public CFContext getContext() {
         return checker;
     }
+
+    /** Method to reflectively set vputil. By default, this method gets the name of type system from
+    * the canonical name of "this", and concatenate it with "VPUtil". If the target class can be
+    * instantiated successfully, it's used. Otherwise, FrameworkVPUtil is returned. Note that
+    * FrameworkVPUtil deosn't perform any viewpoint adaptation function. It's only the base class
+    * for all VPUtil classes of concrete type system. If the concrete type system has a different
+    * naming mechanism, then developer of that type system should override this method to instantiate
+    * it correctly. Otherwise, FrameworkVPUtil is returned but NO viewpoint adaptation is performed.
+    * See {@link FrameworkVPUtil} for more information.
+    * @return GenericVPUtil newly created
+    */
+   protected GenericVPUtil<?> createVPUtil() {
+       // Reflectively lookup subclass of vputil using type factory naming convention
+       String clazzName =
+               this.getClass().getCanonicalName().replace("AnnotatedTypeFactory", "") + "VPUtil";
```

Comment:
Couldn't you replace the ATF with the VPUtil, instead of appending it afterwards?